### PR TITLE
PAINTROID 485 Flutter: Create Onboarding Screen

### PR DESCRIPTION
**Pull request by CodeChamp-SS**
Ticket: https://jira.catrob.at/browse/PAINTROID-485

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -61,13 +77,22 @@ class PocketPaintApp extends StatelessWidget {
...
               case "/":
                 return MaterialPageRoute(
-                  builder: (context) =>
-                      const LandingPage(title: "Pocket Paint"),
+                  builder: (context) => showOnboardingPage
+                      ? const OnboardingPage(
+                          navigateTo: LandingPage(title: 'Pocket Paint'),
+                        )
+                      : const LandingPage(title: 'Pocket Paint'),
+                  // const LandingPage(title: "Pocket Paint"),
```
Comment (juliajulie95) [inline lib/main.dart]: delete

File: lib/main.dart [hunk 2]
```diff
@@ -22,11 +24,25 @@ void main() async {
         error: record.error,
         stackTrace: record.stackTrace);
   });
-  runApp(const ProviderScope(child: PocketPaintApp()));
+
+  WidgetsFlutterBinding.ensureInitialized();
+  final prefs = await SharedPreferences.getInstance();
+  final onBoardingPage = prefs.getBool('onboardingPage') ?? false;
```
Comment (juliajulie95) [inline lib/main.dart]: rename to "showOnboarding" please, also where you save it

File: lib/ui/bottom_control_navigation_bar.dart [hunk 3]
```diff
@@ -38,18 +38,18 @@ class BottomControlNavigationBar extends StatelessWidget {
...
-            icon: const _BottomBarIcon(asset: 'assets/svg/ic_layers.svg'),
+            icon: const BottomBarIcon(asset: 'assets/svg/ic_layers.svg'),
           ),
         ],
       ),
     );
   }
 }
 
-class _BottomBarIcon extends StatelessWidget {
```
Comment (juliajulie95) [inline lib/ui/bottom_control_navigation_bar.dart:49]: Make new file for this, if now used somewhere else too

File: lib/ui/onboarding_page.dart [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+import 'package:paintroid/ui/onboarding_screens/screen3.dart';
+import 'package:paintroid/ui/onboarding_screens/screen4.dart';
+import 'package:paintroid/ui/onboarding_screens/screen5.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+import 'package:smooth_page_indicator/smooth_page_indicator.dart';
+
+import 'onboarding_screens/screen2.dart';
+
+class OnboardingPage extends StatefulWidget {
+  final Widget? navigateTo;
```
Comment (juliajulie95) [inline lib/ui/onboarding_page.dart:14]: I think it would be cleaner if you hand over a VoidCallback here from outside of the class to call in here.
Let me know if I should explain further

File: lib/ui/onboarding_page.dart [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+  State<OnboardingPage> createState() => _OnboardingPageState();
+}
+
+class _OnboardingPageState extends State<OnboardingPage> {
+  final PageController _controller = PageController();
+  bool _isLastPage = false;
+
+  @override
+  void initState() {
+    SystemChrome.setEnabledSystemUIMode(
```
Comment (juliajulie95) [inline lib/ui/onboarding_page.dart:28]: why is this needed?

File: lib/ui/onboarding_screens/screen2.dart [hunk 6]
```diff
@@ -0,0 +1,149 @@
+import 'package:flutter/material.dart';
+import 'package:paintroid/ui/onboarding_page_app_bar.dart';
+import 'package:paintroid/ui/onboarding_page_bottom_nav_bar.dart';
+
+import '../../workspace/src/ui/drawing_canvas.dart';
+import '../bottom_control_navigation_bar.dart';
```
Comment (juliajulie95) [inline lib/ui/onboarding_screens/screen2.dart]: Please check all files for this - imports should not be relative, import as package please

File: lib/ui/onboarding_screens/screen3.dart [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+
+  void watercolor() => toolPressed(13);
+
+  void smudge() => toolPressed(14);
+
+  void clipArea() => toolPressed(15);
+
+  List<BottomNavigationBarItem> _getBottomNavigationBarItems(int l, int r) {
+    List<BottomNavigationBarItem> items = [];
+    for (int i = l; i <= r; i++) {
```
Comment (juliajulie95) [inline lib/ui/onboarding_screens/screen3.dart]: I think this would be more flutter-like if we used a ListViewbuilder

File: lib/ui/onboarding_screens/screen3.dart [hunk 8]
```diff
@@ -0,0 +1,253 @@
...
+              barItems: _getBottomNavigationBarItems(8, 11),
+            ),
+          ),
+          Container(
+            padding: const EdgeInsets.symmetric(vertical: 10),
+            color: lightColorScheme.surface,
+            child: OnboardingPageBottomNavigationBar(
+              onPressedFunctions: [pipette, watercolor, smudge, clipArea],
+              barItems: _getBottomNavigationBarItems(12, 15),
+            ),
```
Comment (juliajulie95) [inline lib/ui/onboarding_screens/screen3.dart]: These are all the same, make a class and reuse it please

**Review by juliajulie95 — CHANGES_REQUESTED** [review]
I added some suggestions, please have a look :)

File: lib/ui/onboarding_page.dart (hunk 5 above)
Comment (CodeChamp-SS) [reply lib/ui/onboarding_page.dart:28]: this is required for showing the onboarding page in the full screen

File: lib/ui/onboarding_screens/screen3.dart (hunk 7 above)
Comment (CodeChamp-SS) [reply lib/ui/onboarding_screens/screen3.dart]: I think ListViewbuilder will be for creating a ListView, rather we can use the List.generate() method

File: lib/ui/bottom_control_navigation_bar.dart (hunk 3 above)
Comment (CodeChamp-SS) [reply lib/ui/bottom_control_navigation_bar.dart:49]: done

File: lib/ui/onboarding_screens/screen3.dart (hunk 8 above)
Comment (CodeChamp-SS) [reply lib/ui/onboarding_screens/screen3.dart]: nice catch, made a new class for this

File: lib/ui/onboarding_page.dart (hunk 4 above)
Comment (CodeChamp-SS) [reply lib/ui/onboarding_page.dart:14]: ah.. you mean a function for the navigation part in the `finish()` method, right?

Comment (juliajulie95) [reply lib/ui/onboarding_page.dart:14]: You declare the function wherever the class is called and hand it over as a boidcallback, then you just call the voidcallback :)

File: lib/ui/onboarding_screens/screen3.dart (hunk 7 above)
Comment (juliajulie95) [reply lib/ui/onboarding_screens/screen3.dart]: Okay, then lets try that :)

**Review by juliajulie95 — APPROVED** [review]
Thank you!
